verilog/tools/ls/verilog-language-server.cc

```diff
@@ -124,35 +162,42 @@ void VerilogLanguageServer::PrintStatistics() const {
 }
 
 verible::lsp::InitializeResult VerilogLanguageServer::InitializeRequestHandler(
-    const nlohmann::json &params) const {
-  // Ignore passed client capabilities from params right now,
-  // just announce what we do.
-  verible::lsp::InitializeResult result;
-  result.serverInfo = {
-      .name = "Verible Verilog language server.",
-      .version = verible::GetRepositoryVersion(),
-  };
-  result.capabilities = {
-      {
-          "textDocumentSync",
-          {
-              {"openClose", true},  // Want open/close events
-              {"change", 2},        // Incremental updates
-          },
-      },
-      {"codeActionProvider", true},               // Autofixes for lint errors
-      {"documentSymbolProvider", true},           // Symbol-outline of file
-      {"documentRangeFormattingProvider", true},  // Format selection
-      {"documentFormattingProvider", true},       // Full file format
-      {"documentHighlightProvider", true},        // Highlight same symbol
-      {"diagnosticProvider",                      // Pull model of diagnostics.
-       {
-           {"interFileDependencies", false},
-           {"workspaceDiagnostics", false},
-       }},
-  };
+    const verible::lsp::InitializeParams &p) {
+  // set VerilogProject for the symbol table, if possible
+  if (!p.rootUri.empty()) {
+    absl::string_view path = verilog::LSPUriToPath(p.rootUri);
+    if (path.empty()) {
+      LOG(ERROR) << "Unsupported rootUri in initialize request:  " << p.rootUri
+                 << std::endl;
+      path = p.rootUri;
+    }
+    ConfigureProject(path);
+  } else if (!p.rootPath.empty()) {
+    ConfigureProject(p.rootPath);
+  } else {
+    LOG(WARNING) << "Could not configure Verilog Project root";
+    ConfigureProject();
+  }
+  return GetCapabilities();
+}
 
-  return result;
+void VerilogLanguageServer::ConfigureProject(absl::string_view project_root) {
+  std::string proj_root = {project_root.begin(), project_root.end()};
+  if (proj_root.empty()) {
+    proj_root = verible::file::Dirname(FindFileList(".")).data();
```

Comment:
data() just returns the pointer to the underlying data, so it is not null-terminated. So this will not return what you think it does.
Instead, if you need a string, convert it accordingly `std::string(verible::file::Dirname(FindFileList("."))`.

Reply:
This explains the issue from above, `/home/hzeller/tt/verible.filelist/bar.sv: Not a directory`